lib/kino/smart_cell/chart_builder.ex

```diff
@@ -0,0 +1,206 @@
+defmodule Kino.SmartCell.ChartBuilder do
+  use Kino.JS, assets_path: "lib/assets/chart_builder"
+  use Kino.JS.Live
+  use Kino.SmartCell, name: "Chart builder"
+
+  @as_int ["width", "height"]
+  @as_atom ["data", "chart", "x_axis_type", "y_axis_type", "color_type"]
+
+  @impl true
+  def init(attrs, ctx) do
+    fields = %{
+      "chart" => attrs["chart"] || "bar",
+      "width" => attrs["width"] || "",
+      "height" => attrs["height"] || "",
+      "x_axis" => attrs["x_axis"] || "",
+      "y_axis" => attrs["y_axis"] || "",
+      "x_axis_type" => attrs["x_axis_type"] || "",
+      "y_axis_type" => attrs["y_axis_type"] || "",
+      "color" => attrs["color"] || "",
+      "color_type" => attrs["color_type"] || "",
+      "data" => attrs["data"] || ""
+    }
```

Comment:
It would be okay by me, but I honestly don't see the benefits. We already have consistency since we have empyt strings for JS and `nil` for Elixir, and the conversion happens only on `to_source`. I also think it would lead us to have comparisons with  `=== "nil"` on JS's side which doesn't hurt but seems a bit odd... Unless I'm missing something here 🤔

Reply:
Ah, I wrote that before I realised that `convert_field` doesn't happen on change, but before code generation.

I'm mostly wondering about the types we should have in `attrs` (because that's what we persist and it's hard to change later on). It may be convenient to just keep the values coming from JS, which are pretty much always strings, but I think keeping the parsed values is cleaner. So for fields that are selects either value or `nil`, for numbers either integer or `nil`, for text fields always string. In other words, I'd lean towards parsing the data on change, though I don't have a strong reason other than it _feels_ better to me and I believe that's what we do in the other cells :)

> I also think it would lead us to have comparisons with `=== "nil"`

`nil` becomes `null` on JS side :)

Reply:
Done 🐈